src/goto-instrument/contracts/memory_predicates.cpp

```diff
@@ -137,7 +137,7 @@ void is_fresh_baset::update_ensures(goto_programt &ensures)
 
 array_typet is_fresh_baset::get_memmap_type()
 {
-  return array_typet(c_bool_typet(8), infinity_exprt(size_type()));
```

Comment:
Do we really need to propagate this to instrumentation code ? This code does not represent anything running on the actual processor. Is `c_bool_typet(8)` deprecated ?

Reply:
I'm ok with an arbitrary type, but we must not have "8" as a magic number. I assumed this was "8" in the first place so as to have byte granularity? Was I wrong?

Reply:
My usual assumption is that architecture should be well defined at the point of running the front end. This is because the C front end runs an external C pre-processor which can have architecture specific macro expansions. As `goto-instrument` runs after the front end, it is operating on an architecture specific program form. Note for example that the `c_bool_type()` construction function in `src/util/c_types.cpp` references the same config field.